esphome/components/remote_base/__init__.py

```diff
@@ -995,3 +996,108 @@ async def panasonic_action(var, config, args):
     cg.add(var.set_address(template_))
     template_ = await cg.templatable(config[CONF_COMMAND], args, cg.uint32)
     cg.add(var.set_command(template_))
+
+
+# Midea
+MideaData, MideaBinarySensor, MideaTrigger, MideaAction, MideaDumper = declare_protocol(
+    "Midea"
+)
+MideaRawAction = ns.class_("MideaRawAction", RemoteTransmitterActionBase)
+MideaFollowMeAction = ns.class_("MideaFollowMeAction", RemoteTransmitterActionBase)
+MideaDisplayToggleAction = ns.class_(
+    "MideaDisplayToggleAction", RemoteTransmitterActionBase
+)
+MideaSwingStepAction = ns.class_("MideaSwingStepAction", RemoteTransmitterActionBase)
+MIDEA_RAW_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_CODE): cv.All(
+            [cv.Any(cv.hex_uint8_t, cv.uint8_t)],
+            cv.Length(min=5, max=5),
+        ),
+        cv.GenerateID(CONF_CODE_STORAGE_ID): cv.declare_id(cg.uint8),
+    }
+)
+
+
+@register_binary_sensor("midea", MideaBinarySensor, MIDEA_RAW_SCHEMA)
+def midea_binary_sensor(var, config):
+    code_ = config[CONF_CODE]
+    arr = cg.progmem_array(config[CONF_CODE_STORAGE_ID], code_)
+    cg.add(var.set_data(arr))
+    cg.add(var.set_len(len(code_)))
+
+
+@register_trigger("midea", MideaTrigger, cg.std_vector.template(cg.int32))
+def midea_trigger(var, config):
+    pass
+
+
+@register_dumper("midea", MideaDumper)
+def midea_dumper(var, config):
+    pass
+
+
+@register_action(
+    "midea_raw",
+    MideaRawAction,
+    MIDEA_RAW_SCHEMA,
+)
+async def midea_raw_action(var, config, args):
+    code_ = config[CONF_CODE]
+    if cg.is_template(code_):
+        template_ = await cg.templatable(code_, args, cg.std_vector.template(cg.uint8))
+        cg.add(var.set_template(template_))
+    else:
+        arr = cg.progmem_array(config[CONF_CODE_STORAGE_ID], config[CONF_CODE])
+        cg.add(var.set_data(arr))
+
+
+# Midea FollowMe action
+CONF_BEEPER = "beeper"
+MIDEA_FOLLOW_ME_MIN = 0
+MIDEA_FOLLOW_ME_MAX = 37
+MIDEA_FOLLOW_ME_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_TEMPERATURE): cv.All(
+            cv.int_range(MIDEA_FOLLOW_ME_MIN, MIDEA_FOLLOW_ME_MAX),
+        ),
+        cv.Optional(CONF_BEEPER, default=False): cv.All(
+            cv.boolean,
+        ),
+    }
+)
```

Comment:
These can also just be set to templatable

```suggestion
MIDEA_FOLLOW_ME_SCHEMA = cv.Schema(
    {
        cv.Required(CONF_TEMPERATURE): cv.templatable(
            cv.int_range(MIDEA_FOLLOW_ME_MIN, MIDEA_FOLLOW_ME_MAX),
        ),
        cv.Optional(CONF_BEEPER, default=False): cv.templatable(cv.boolean),
    }
)
```